src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,972 @@
+            var referencedSymbols = progress.GetReferencedSymbols();
+
+            // Ordering by descending to sort invocations by starting span to account for nested invocations
+            var referencedLocations = referencedSymbols.SelectMany(referencedSymbol => referencedSymbol.Locations)
+                .Distinct().Where(reference => !reference.IsImplicit)
+                .OrderByDescending(reference => reference.Location.SourceSpan.Start);
+
+            // Adding the original document to ensure that it will be seen again when processing the call sites
+            // in order to update the original expression and containing method.
+            methodCallSitesBuilder.Add(document, new List<SyntaxNode>());
+
+            foreach (var refLocation in referencedLocations)
+            {
+                // Does not support cross-language references currently
+                if (refLocation.Document.Project.Language == document.Project.Language)
+                {
+                    var reference = refLocation.Location.FindNode(cancellationToken).GetRequiredParent();
+                    while (reference is not (TObjectCreationExpressionSyntax or TInvocationExpressionSyntax))
+                    {
+                        reference = reference.GetRequiredParent();
+                    }
+
+                    // Only adding items that are of type InvocationExpressionSyntax or TObjectCreationExpressionSyntax
+                    var invocationOrCreation = reference as TObjectCreationExpressionSyntax ?? (SyntaxNode?)(reference as TInvocationExpressionSyntax);
+                    if (invocationOrCreation is null)
+                    {
+                        continue;
+                    }
+
+                    if (!methodCallSitesBuilder.TryGetValue(refLocation.Document, out var list))
+                    {
+                        list = new List<SyntaxNode>();
+                        methodCallSitesBuilder.Add(refLocation.Document, list);
+                    }
+
+                    list.Add(invocationOrCreation);
+                }
+            }
+
+            return methodCallSitesBuilder.ToImmutable();
+        }
+
+        /// <summary>
+        /// If the parameter is optional and the invocation does not specify the parameter, then
+        /// a named argument needs to be introduced.
+        /// </summary>
+        private static SeparatedSyntaxList<SyntaxNode> AddArgumentToArgumentList(
+            SeparatedSyntaxList<SyntaxNode> invocationArguments, SyntaxGenerator generator,
+            SyntaxNode newArgumentExpression, string parameterName, int insertionIndex, bool named)
+        {
+            var argument = named
+                ? generator.Argument(parameterName, RefKind.None, newArgumentExpression)
+                : generator.Argument(newArgumentExpression);
+            return invocationArguments.Insert(insertionIndex, argument.WithAdditionalAnnotations(Simplifier.Annotation));
```

Comment:
what's the simplifier annotation for?  can you doc in code?

Reply:
at some point I thought I needed it, I do not. removed